modules/core/src/test/java/org/apache/ignite/cdc/ChangeDataCaptureOrderTest.java

```diff
@@ -0,0 +1,250 @@
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.cdc.ChangeDataCapture;
+import org.apache.ignite.internal.processors.cache.CacheConflictResolutionManager;
+import org.apache.ignite.internal.processors.cache.CacheObject;
+import org.apache.ignite.internal.processors.cache.CacheObjectImpl;
+import org.apache.ignite.internal.processors.cache.CacheObjectValueContext;
+import org.apache.ignite.internal.processors.cache.GridCacheManagerAdapter;
+import org.apache.ignite.internal.processors.cache.IgniteInternalCache;
+import org.apache.ignite.internal.processors.cache.KeyCacheObject;
+import org.apache.ignite.internal.processors.cache.KeyCacheObjectImpl;
+import org.apache.ignite.internal.processors.cache.dr.GridCacheDrInfo;
+import org.apache.ignite.internal.processors.cache.version.CacheVersionConflictResolver;
+import org.apache.ignite.internal.processors.cache.version.GridCacheVersion;
+import org.apache.ignite.internal.processors.cache.version.GridCacheVersionConflictContext;
+import org.apache.ignite.internal.processors.cache.version.GridCacheVersionedEntryEx;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.plugin.AbstractCachePluginProvider;
+import org.apache.ignite.plugin.AbstractTestPluginProvider;
+import org.apache.ignite.plugin.CachePluginContext;
+import org.apache.ignite.plugin.CachePluginProvider;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.jetbrains.annotations.Nullable;
+import org.junit.Test;
+
+import static org.apache.ignite.cdc.ChangeDataCaptureSelfTest.KEYS_CNT;
+import static org.apache.ignite.cdc.ChangeDataCaptureSelfTest.WAL_ARCHIVE_TIMEOUT;
+import static org.apache.ignite.cdc.ChangeDataCaptureSelfTest.addAndWaitForConsumption;
+import static org.apache.ignite.cdc.ChangeDataCaptureSelfTest.cdcConfig;
+import static org.apache.ignite.cluster.ClusterState.ACTIVE;
+import static org.apache.ignite.configuration.WALMode.FSYNC;
+import static org.apache.ignite.testframework.GridTestUtils.runAsync;
+import static org.apache.ignite.testframework.GridTestUtils.waitForCondition;
+
+/** */
+public class ChangeDataCaptureOrderTest extends GridCommonAbstractTest {
+    /** */
+    public static final String FOR_OTHER_DR_ID = "for-other-dr-id";
+
+    /** */
+    public static final byte OTHER_DR_ID = 2;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        int segmentSz = 10 * 1024 * 1024;
+
+        cfg.setDataStorageConfiguration(new DataStorageConfiguration()
+            .setChangeDataCaptureEnabled(true)
+            .setWalMode(FSYNC)
```

Comment:
CDC should work even if BACKGROUND is set?

Reply:
Yes. Other WAL modes checked in `ChangeDataCaptureSelfTest`.

Reply:
Since it's not required here, could we get rid of this mention?

Reply:
done